### ISSUE TITLE
Move procedural code from Slim.php to a better place

### ISSUE THREAD

**Issue by ralphschindler** [slimphp/Slim#266]
The following code inside Slim.php:

``` php
<?php
//Ensure PHP session IDs only use the characters [a-z0-9]
ini_set('session.hash_bits_per_character', 4);
ini_set('session.hash_function', 0);

//Slim's Encryted Cookies rely on libmcyrpt and these two constants.
//If libmycrpt is unavailable, we ensure the expected constants
//are available to avoid errors.
if ( !defined('MCRYPT_RIJNDAEL_256') ) {
    define('MCRYPT_RIJNDAEL_256', 0);
}
if ( !defined('MCRYPT_MODE_CBC') ) {
    define('MCRYPT_MODE_CBC', 0);
}

//This determines which errors are reported by PHP. By default, all
//errors (including E_STRICT) are reported.
error_reporting(E_ALL | E_STRICT);

//This tells PHP to auto-load classes using Slim's autoloader; this will
//only auto-load a class file located in the same directory as Slim.php
//whose file name (excluding the final dot and extension) is the same
//as its class name (case-sensitive). For example, "View.php" will be
//loaded when Slim uses the "View" class for the first time.
spl_autoload_register(array('Slim', 'autoload'));

//PHP 5.3 will complain if you don't set a timezone. If you do not
//specify your own timezone before requiring Slim, this tells PHP to use UTC.
if ( @date_default_timezone_set(date_default_timezone_get()) === false ) {
    date_default_timezone_set('UTC');
}
```

Should be relocated based on its purpose:
- ini_set's for the session specific stuff should be moved to Slim_Session.
- the mcrypt stuff should be moved to the cookies class
- The error_reporting() should be moved to the default `index.php` script
- The autoloader should be moved to `index.php`
- The datatimezone should be moved to `index.php`

The primary goals of these changes would be to make more portable codebase out of Slim so that it plays nice with 3rd party libraries, works well inside a Phar, and in general makes Slim more of a "library" with the primary use case being the index.php (as a starting point)

Specific situations this would help:
- "I have my own autoloader"
- "I have to use a different error_reporting setting for some other code I am using"
- ...

Thoughts?

**Comment by codeguy** [comment]
I agree completely. This has already been done in the develop branch and is scheduled to be included in version 1.6.0 when that is released soon.

**Comment by ralphschindler** [comment]
Out of curiosity, how do you feel about this:
https://github.com/codeguy/Slim/blob/release-1.6.3/Slim/Slim.php#L34-L56
and moving that out into the index.php?

-ralph
